### ISSUE TITLE
Implémenter les profils de taxons dans l'application mobile

### ISSUE THREAD

**Issue by DonovanMaillard** [PnX-SI/gn_mobile_occtax#196]
**Type d'amélioration**
Fonctionnalité

**Proposition**
Depuis la V2 de Occtax-mobile, un certains nombre de développements ont été faits pour aligner autant que possible les fonctionnalités du module entre le web et le mobile. 

Parmi les deltas restant, il serait intéressant d'implémenter dans le mobile les profils de taxons qui sont mis en place sur le serveur. Comme pour les couleurs de taxons, il s'agirait d'une information optionnelle, qui permettrait de contextualiser la donnée et d'alerter l'utilisateur lors d'une saisie "hors connaissances déjà disponibles dans l'instance". 

Pour cela, il faudrait : 
- Ajouter un paramètre pour activer ou non les profils
- Ajouter la récupération des profils au processus de synchronisation des données (dans le bloc 'données optionnelles')
- Implémenter les contrôles et les alertes de saisie lorsque la saisie n'est pas conforme aux profils

Non financé ni planifié à l'heure actuelle, on reste disponible pour accompagner tout volontaire qui voudrait porter cette évolution.

**Comment by TheoLechemia** [comment]
Je viens de percuter sur un truc. Le calcul des profils nécessite l'altitude. Si celle-ci est absente, la route renvoie une 404. Le mobile ne calculant pas cette info, on va être bloqué.. 
Deux options :
- on se passe de l'altitude pour interroger le profile (on peut renvoyer une info fausse)
- comme c'est une fonctionnalité en ligne, on interroge un webservice nous renvoyant cette info (IGN) ?

**Comment by DonovanMaillard** [comment]
Sur le 2eme point, on est plutôt partis du principe que c'est une fonctionnalité embarquée comme tout le reste des données et des fonctionnalités, on aurait donc une récupération des données - comme pour les couleurs des taxons - lancée de manière périodique selon le même paramètre que pour les couleurs. 

De là, on a soit la possibilité de charger un mnt en local (houtch!!). Soit on fait en sorte que l'altitude soit facultative (on se limite au zonage et aux dates). 

Soit dans ce cas on adopte un fonctionnement un peu différent mais nouveau, et qui dans l'idée me dérange : c'est lors de la syncro (forcément en ligne), qu'on va checker toutes les données avant de les poster. Mais dans ce cas, on a des alertes au moment de la synchro, hors contexte de saisie, à un moment ou on a pas forcément le temps ou l'envie d'aller controler ses saisies.... 

Dans l'idée, je dirais qu'on peut se satisfaire d'un profil qui ignore les altitudes, donc en allant comparer le zonage et la date et en ne mettant des alertes que sur ce critère là ? 


Si l'option préférée est de garder cette fonctionnalité en mode "online", il faut aligner les couleurs là dessus aussi pour s'économiser cette synchro et être explicite quand ca fonctionne ou non, que l'utilisateur n'attende pas une info qui n'arrivera pas... mais bon. Je préfère considérer que l'appli peut fonctionner offline sans les altitudes.

**Comment by camillemonchicourt** [comment]
Bien vu, à voir si l'intégration des profils est jouable/pertinente.

**Comment by DonovanMaillard** [comment]
Oui, et réétudier cette possibilité du online pour les informations facultatives. C'est pas forcément idéal, mais pourquoi pas...
